### ISSUE TITLE
Slow start/restart/coming from background

### ISSUE THREAD

**Issue by telmich** [RocketChat/Rocket.Chat.Cordova#130]
Hello!

I am using RocketChat android application and it start slowly.
If the app gets to background when I want it again in front then it
acts as it connects and fetches everything again, like when you start it for first time,
the only difference is that it does not ask for login and password.
So, when I use app and I am writing a message, and I want to see something on, for example, web I have to switch to browser and then when I want to get back to RC app it starts again with connecting/fetching, I don't know, but I have to wait for it, and usually I lost message in input field I didn't yet send (I want to paste something from browser, for example).
Can you do something about it?

(follow up from https://github.com/RocketChat/Rocket.Chat.Android/issues/17)

@greendeath and @darko-poljak have the same issue as I have
##

**Comment by telmich** [comment]
Thanks for the update, @graywolf336!

**Comment by Syirrus** [comment]
I'm having the same issue with cordova on iOS and on Android. It has nothing to do with Meteor.reconnect(); either. I have changed the value to prevent that from having to reconnect after only a minute. Any ideas? The app probably takes 10 seconds to become responsive once you have focus on it from it being in the background. Did you get this fixed @telmich ?

**Comment by telmich** [comment]
@Syirrus unfortunately not - the biggest pain point with rocket chat is the mobile client at the moment

**Comment by Syirrus** [comment]
@telmich Yeah this is a big issue. I'm not convinced it is entirely something in Cordova, although things seem to work fine in the browser. Perhaps it's some unnecessary stuff occurring on the server. I don't I'm just guessing out of frustration. Things seemed faster in earlier builds of rocket.chat (server) part.

**Comment by Syirrus** [comment]
@telmich I might have found some things to speed the app up.
1. Try updating your version of crosswalk to 21+
2. Increase the timeout value for which RC Cordova disconnects when in the background. 
3. Update ionic-plugin-keyboard to the latest. (Before updating this I noticed the keyboard would focus and unfocus every time I went into the app. Now the keyboard doesn't do that anymore which might be another reason why things were slow)

These changes **significantly** improved the response of the app, though it still has stutters a tiny bit.

**Comment by TheReal1604** [comment]
@Syirrus are this server-side settings? Or in the Cordova Mobile App so i have to repackage the app.

**Comment by Syirrus** [comment]
@TheReal1604 both. 
Updating crosswalk to 21+ will be done in Cordova Rocket.Chat in config.xml
Updating ionic-plugin-keyboard should be done in cordova.
Increasing the timeout value for when Cordova clients disconnect needs to happen on the server.

**Comment by kevincolten** [comment]
> Increasing the timeout value for when Cordova clients disconnect needs to happen on the server.

@Syirrus Can you explain where on the server you'd configure the timeout value?

**Comment by Syirrus** [comment]
@kevincolten You can update the timeout value in:
/packages/rocketchat-ui/lib/cordova/user-state.js

It helps on Android. However, resuming from the background still pauses a bit.

**Comment by pshute** [comment]
I'm wondering if I have the same problem in iOS. If I use other apps for a while then return to Rocket.Chat, it will often start up from scratch, and go to the home screen. It usually stays logged in, but I have seen it log me out, perhaps after an app update, can't remember.

I assume this is to do with the app being unloaded from RAM by iOS. This iPad only has 1GB RAM, so more recent devices may not encounter the problem as much.

It would be great if it could restore the state of the app and restart in the channel I was in, with the messages scrolled to the same position.

Is it too much to ask for it to also restore a message I was part way through typing? A typical scenario is I switch to Safari to research something for a message I'm typing, then when I return later it's all gone.

It would probably be a good idea to retain the current behaviour when restarting after the app has been forced closed by the user, in case they're trying to recover from a problem.

I'm also wondering if this is related to #140 .

**Comment by choois** [comment]
Noticed the same thing on Android.
Every time I reactivate the app from the background by clicking on the icon, the app resets to initial state.
